app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -204,56 +249,24 @@ class LuciBuildService {
         'repo_name': slug.name,
         'user_agent': 'flutter-cocoon',
       };
-      int checkRunId;
-      if (checkSuiteEvent != null || config.githubPresubmitSupportedRepo(slug)) {
-        log.info('Creating check run for PR: $prNumber, Commit: $commitSha, Slug: $slug');
-        final github.CheckRun checkRun = await githubChecksUtil.createCheckRun(
-          config,
-          slug,
-          builder,
-          commitSha,
-        );
-        userData['check_run_id'] = checkRun.id;
-        checkRunId = checkRun.id;
-      }
-      requests.add(
-        Request(
-          scheduleBuild: ScheduleBuildRequest(
-            builderId: builderId,
-            tags: <String, List<String>>{
-              'buildset': <String>['pr/git/$prNumber', 'sha/git/$commitSha'],
-              'user_agent': const <String>['flutter-cocoon'],
-              'github_link': <String>['https://github.com/${slug.fullName}/pull/$prNumber'],
-              if (checkRunId != null) 'github_checkrun': <String>[checkRunId.toString()],
-            },
-            properties: <String, String>{
-              'git_url': 'https://github.com/${slug.owner}/${slug.name}',
-              'git_ref': 'refs/pull/$prNumber/head',
-            },
-            notify: NotificationConfig(
-              pubsubTopic: 'projects/flutter-dashboard/topics/luci-builds',
-              userData: json.encode(userData),
-            ),
-          ),
-        ),
-      );
+      requestFutures.add(_createBuildRequest(checkSuiteEvent, slug, prNumber, commitSha, builder, builderId, userData));
     }
-    const RetryOptions r = RetryOptions(
-      maxAttempts: 3,
-      delayFactor: Duration(seconds: 2),
-    );
-    BatchResponse batchResponse;
-    log.debug('Making BatchRequest with ${requests.length} requests');
-    await r.retry(
-      () async {
-        batchResponse = await buildBucketClient.batch(BatchRequest(requests: requests));
-      },
-      retryIf: (Exception e) => e is BuildBucketException,
-    );
-    for (Response response in batchResponse.responses) {
-      if (response.error?.code != 0) {
-        log.warning('BatchResponse error: $response');
-        continue;
+    final List<Request> requests = await Future.wait(requestFutures);
+    List<BatchResponse> batchResponseList;
+    final List<Future<BatchResponse>> futureBatchResponses = <Future<BatchResponse>>[];
+    final Iterable<List<Request>> requestPartitions = await shard(requests, 1);
```

Comment:
Is this intentional to have batch size 1? This will incur too many buildbucket API calls, have we checked this comply with QPS requirement from buildbucket side?

Reply:
I made this a configuration value with a value of 5. We don't have a way to measure the flutter QPS but this is internal traffic and minimal compared to other clients like fuchsia.